research/conditional/moe_layers/continuous_moe.py

```diff
@@ -46,17 +46,17 @@ def __post_init__(self):
                 f"expert_size is None, setting it to dff // n_experts = {self.dff // self.n_experts}"
             )
             self.expert_size = self.dff // self.n_experts
-        self.init_parameters()
+        self.init_core_parameters()
+        self.init_additional_parameters()
```

Comment:
Why such distinction?

Reply:
It allows child layers to introduce new parameters, without copypasting init of the existing parameters.